### PR TITLE
Instruct ServiceMonitor to connect using https for controller

### PR DESCRIPTION
**Pull request by jstewart612**
The controller metrics endpoint serves over https using a self-signed cert by default in this chart so correct the ServiceMonitor to reflect.

### CODE REVIEW

**Comment by jstewart612** [comment]
Okay it only does this if metrics.proxy.enabled is set to true and if so then you need this and mTLS handling.  This needs more work.

**Comment by mumoshu** [comment]
@jstewart612 Hey! Thanks. How can we move forward? Should I just enclose the below part within a `{{ if }} {{ end }}` block that is enabled only when you enabled metrics proxy?

```
      scheme: https
      tlsConfig:
        insecureSkipVerify: true
```

**Comment by jstewart612** [comment]
@mumoshu yes but an additional problem is that when https is enabled, it appears to function using mTLS so I hadn't yet researched the schema of ServiceMonitor yet to find the mTLS options to specify.

**Comment by mumoshu** [comment]
@jstewart612 Hey! Thanks, that makes sense.

I've googled about the matter a bit and found a potential solution at https://github.com/operator-framework/operator-sdk/issues/4764#issuecomment-820344706.

Unfortunately, I have other works I have to prioritize higher now. I would appreciate it very much if you could test it and include the changes into this PR if it worked 🙏

File: charts/actions-runner-controller/templates/controller.metrics.serviceMonitor.yaml [hunk 1]
```diff
@@ -9,6 +9,9 @@ spec:
   endpoints:
     - path: /metrics
       port: metrics-port
+      scheme: https
```
Comment (mumoshu) [inline charts/actions-runner-controller/templates/controller.metrics.serviceMonitor.yaml:12]: @jstewart612 Hey. Awesome @AlexDHoffer has figured out that we can add `bearerTokenFile` to the endpoint to make it work!

```suggestion
      bearerTokenFile: /var/run/secrets/kubernetes.io/serviceaccount/token
      scheme: https
```

Would you mind confirming it and wrap this change into a `if end` block to make it configurable? Once done I'll quickly merge this

**Comment by jstewart612** [comment]
@mumoshu added.  Should be good to go now.  Works for me as well as the user that reported that.

File: charts/actions-runner-controller/Chart.yaml [hunk 2]
```diff
@@ -15,7 +15,7 @@ type: application
 # This is the chart version. This version number should be incremented each time you make changes
 # to the chart and its templates, including the app version.
 # Versions are expected to follow Semantic Versioning (https://semver.org/)
-version: 0.12.2
+version: 0.12.3
```
Comment (mumoshu) [inline charts/actions-runner-controller/Chart.yaml:18]:
```suggestion
version: 0.12.4
```

**Review by mumoshu — APPROVED** [review]
LGTM. Thanks a lot for your contribution @jstewart612 (and @AlexDHoffer for support) ☺️

**Pull request merged**